src/Workspaces/Core/Desktop/Workspace/Host/Mef/MefV1HostServices.cs

```diff
@@ -39,18 +43,28 @@ public static MefV1HostServices Create(ExportProvider exportProvider)
             return new MefV1HostServices(exportProvider);
         }
 
-        public static MefV1HostServices Create(IEnumerable<System.Reflection.Assembly> assemblies)
+        public static MefV1HostServices Create(IEnumerable<Assembly> assemblies)
         {
             if (assemblies == null)
             {
                 throw new ArgumentNullException(nameof(assemblies));
             }
 
+            if (s_CreationHook != null)
+            {
+                return s_CreationHook(assemblies);
+            }
+
             var catalog = new AggregateCatalog(assemblies.Select(a => new AssemblyCatalog(a)));
             var container = new CompositionContainer(catalog, compositionOptions: CompositionOptions.DisableSilentRejection | CompositionOptions.IsThreadSafe);
             return new MefV1HostServices(container);
         }
 
+        internal static void HookServiceCreation(CreationHook hook)
```

Comment:
> internal static void HookServiceCreation(CreationHook hook) [](start = 7, length = 60)

Should we assert that `s_CreationHook` is `null`? Lacking that seems like we will risk introducing bugs by silently overwriting the previous vaule.

Reply:
➡️ No. In the cleanup step for the test hook, the creation hook is replaced by a non-null hook that produces an informative exception/test failure in the event code attempts to create an ExportProvider after the test exits the isolation boundary.

Reply:
For cases where the overwrite is inteended a `force` parameter could be employed. 


---
In reply to: [175523407](https://github.com/dotnet/roslyn/pull/25558#discussion_r175523407) [](ancestors = 175523407)